rmgpy/molecule/pathfinder.py

```diff
@@ -480,3 +480,51 @@ def is_atom_able_to_lose_lone_pair(atom):
     return (((atom.is_nitrogen() or atom.is_sulfur()) and atom.lone_pairs in [1, 2, 3])
             or (atom.is_oxygen() and atom.lone_pairs in [2, 3])
             or atom.is_carbon() and atom.lone_pairs == 1)
+
+def find_adsorbate_delocalization_paths(atom1):
+    """
+    Find all bidentate adsorbates which have a bonding configuration X-C-C-X.
+    Examples:
+
+    - XCXC, XCHXCH, XCXCH, where X is the surface site
+
+    In this transition atom1 and atom4 are surface sites while atom2 and atom3 are carbon atoms.
+    """
+    cython.declare(paths=list, atom2=Vertex, atom3=Vertex, atom4=Vertex, bond12=Edge, bond23=Edge, bond34=Edge)
+
+    path=[]
+    if atom1.is_surface_site():
+        for atom2, bond12 in atom1.edges.items():
+            if atom2.is_carbon():
+                for atom3, bond23 in atom2.edges.items():
+                    if atom1 is not atom3 and atom3.is_carbon():
+                        for atom4, bond34 in atom3.edges.items():
+                            if atom2 is not atom4 and atom4.is_surface_site():
+                                path.append([atom1, atom2, atom3, atom4, bond12, bond23, bond34])
+                                return path
+    return path
+
+def find_adsorbate_conjugate_delocalization_paths(atom1):
+    """
+    Find all bidentate adsorbates which have a bonding configuration X-C-C-C-X.
+    Examples:
+
+    - XCHCHXCH/XCHCHXC, where X is the surface site
```

Comment:
It's a bit confusing for me that `X` is not always listed at the end. Is this customary?

Reply:
Unfortunately, the notation for adsorbates is a little confusing since it is difficult to represent the adsorbate structure as a string of letters. This is the current notation that we and some other groups use.